src/Model.php

```diff
@@ -960,49 +965,19 @@ public function _unset(string $name)
      */
     public function addCondition($field, $operator = null, $value = null)
     {
-        if (is_array($field)) {
-            $this->conditions[] = [$field];
-
-            return $this;
-            /*
-            $or = $this->persistence->orExpr();
-
-            foreach ($field as list($field, $operator, $value)) {
-                if (is_string($field)) {
-                    $f = $this->getField($field);
-                } elseif ($field instanceof Field) {
-                    $f = $field;
-                }
-
-                $or->where($f, $operator, $value);
-            }
-
-            return $this;
-            */
-        }
-
-        if (is_string($field)) {
-            $f = $this->getField($field);
-        } else {
-            $f = $field;
-        }
-
-        if ($f instanceof Field) {
-            if ($operator === '=' || func_num_args() === 2) {
-                $v = ($operator === '=' ? $value : $operator);
-
-                if (!is_object($v) && !is_array($v)) {
-                    $f->system = true;
-                    $f->default = $v;
-                }
-            }
-        }
-
-        $this->conditions[] = func_get_args();
+        $this->scope()->add(new Model\Scope\CompoundCondition([func_get_args()]));
```

Comment:
I would prefer to not pass `func_get_args()` - the code can not be read easily - do we want to allow the full input here?

Also, this can add an extra level of Scope for simple condition, right? We should probably try to simply the condition to add.

Reply:
This is intentional and for backward compatibility on setting nested OR conditions